utils/common/src/main/kotlin/ProcessCapture.kt

```diff
@@ -84,7 +84,7 @@ class ProcessCapture(
         .redirectOutput(stdoutFile)
         .redirectError(stderrFile)
         .apply {
-            environment().putAll(environment)
+            EnvironmentVariableFilter.filterEnvironment(environment()).putAll(environment)
```

Comment:
Shouldn't the filter be applied on the second `environment` argument?

Reply:
You are probably right, but I am not 100% sure. My thinking was that environment variables that are explicitly provided when the process is created are probably mandatory for the command to function correctly and should therefore not be filtered. From what I have seen, only at few places environment variables are passed, for instance to propagate the GO path or to disable SSL verification in NPM.

Reply:
> environment variables that are explicitly provided when the process is created are probably mandatory for the command to function correctly and should therefore not be filtered

I agree. Could you maybe add a code comment about that? I.e. explain why `environment()` is filtered but `environment` not?

Reply:
Added a comment.